### ISSUE TITLE
Reproduce the results of Transformer-Base on WMT17 En-Zh

### ISSUE THREAD

**Issue by xwgeng** [pytorch/fairseq#1424]
Hi, 

I want to reproduce the results of Transformer-base model, which can achieve about 34 BLEU on WMT17 En->Zh translation. The transformer-base model was trained with the following scripts:

```Bash
fairseq-train \
    data-bin/wmt17_en_zh \
    --source-lang en --target-lang zh \
    --arch transformer_wmt_en_de
    --save-dir checkpoints \
    --ddp-backend=no_c10d \
    --criterion label_smoothed_cross_entropy \
    --optimizer adam --adam-betas '(0.9,0.98)' --clip-norm 0.0 \
    --lr 0.0005 --lr-scheduler inverse_sqrt \
    --min-lr '1e-09' --warmup-updates 4000 \
    --warmup-init-lr '1e-07' --label-smoothing 0.1 \
    --dropout 0.3 --weight-decay 0.0 \
    --log-format 'simple' --log-interval 100 \
    --fixed-validation-seed 7 \
    --max-tokens 8000 \
    --save-interval-updates 10000 \
    --max-update 300000
    --fp16
```
I tested the performance of transformer-base with following script:
```Bash
fairseq-generate \ 
    data-bin/wmt17_en_zh \
    --path checkpoints/checkpoint_best.pt \
    --beam 5 --remove-bpe \
    --batch-size 200 \
```
I trained transform-base model using 8 GPUs with a total batch size 64k. However, it can only obtain 20 BLEU on test set, while using the same training script, the resulted transformer-base model on WMT17 Zh->En can achieve 24 BLEU, comparable to  Hassan et al. (2018)'s implementation.

**Comment by myleott** [comment]
Can you share the training log?

**Comment by xwgeng** [comment]
Thanks for your reply.

The part of training logs is as follows:
![log](https://user-images.githubusercontent.com/7008668/69559246-882f9000-0fe4-11ea-8f1c-3b8eaefef703.png)
At the first epoch, transformer-base model can achieve 16.40 BLEU on WMT17 En->Zh, and after about 90k update steps, it can obtain 20 BLEU. Later, no improvement can be observed.

**Comment by xwgeng** [comment]
I also attempted to train transformer-base model with batch size 128k as the settings in Ghazvininejad et al. (2019)'s [paper](https://www.aclweb.org/anthology/D19-1633.pdf). However, transformer-base trained on WMT17 En->Zh achieved similar results (20 BLEU).

**Comment by huihuifan** [comment]
Are you sure the generation parameters you use (including checkpoint averaging) are the same?

**Comment by xwgeng** [comment]
You mean that the generation parameters for En->Zh and Zh->En are the same? Sure, they are the same. The generation parameters are as follows:
![generation](https://user-images.githubusercontent.com/7008668/69622494-ff643300-107b-11ea-9cec-05462e975d1f.png)

**Comment by xwgeng** [comment]
I have solved this issue. The reason for poor performance on WMT En->Zh is that the BLEU score is calculated based on tokenized Chinese output.

**Comment by suyuzhang** [comment]
Hi @xwgeng 
Could you share your data preprocessing script?
Thanks!

**Comment by xwgeng** [comment]
Sorry for late reply. Please refer to the repo https://github.com/xwgeng/WMT17-scripts @suyuzhang
